pyqldb/driver/qldb_driver.py

```diff
@@ -283,20 +298,23 @@ def _release_session(self, session):
         """
         Release a session back into the pool.
         """
-
-        if not session._is_closed:
-            self._pool.put(session)
-
         self._pool_permits.release()
         self._pool_permits_counter.increment()
         logger.debug('Number of sessions in pool : {}'.format(self._pool.qsize()))
 
-    def _get_session(self):
+        if session is not None and session._is_alive:
```

Comment:
Why the change to put this into the pool after increasing the permits? Will that result in attempting to retrieve items from the pool before they're placed there?

Reply:
Good point, I will update. 

I think that means [Node.JS](https://github.com/awslabs/amazon-qldb-driver-nodejs/blob/master/src/QldbDriver.ts#L210-L214) will also need to update

Reply:
It should be updated, but note that Node likely won't suffer from the problem as it's single-threaded.